src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -207,6 +181,36 @@ public String getScript() {
         return c;
     }
 
+    private String generateLaunchCmd(Jenkins jenkins, ShellController c, FilePath ws, OsType os, String interpreter, String scriptPath, String cookieValue, String cookieVariable) throws IOException, InterruptedException {
+        String logFile = c.getLogFile(ws).toURI().getRawPath();
+        String resultFile = c.getResultFile(ws).toURI().getRawPath();
+        FilePath controlDir = c.controlDir(ws);
+        String outputFile = c.getOutputFile(ws).toURI().getRawPath();
+
+        String launcher_binary = LAUNCHER_PREFIX + os.toString();
+
+        InputStream launcherStream = this.getClass().getResourceAsStream(launcher_binary);
```

Comment:
i'll go ahead and do the docker route, generating the binaries at build time.  Right now, it looks like i'll need to do 3 versions (assuming i can get the windows version to work). probably take up just under 10 megs total for all 3.

Reply:
You can use [`golang`](https://hub.docker.com/_/golang) for this purpose (a specific tag please)! The problem is that then the `Jenkinsfile` needs to specify

```groovy
platforms: ['docker']
```

which means that we would no longer be running _tests_ on Windows build nodes, because we do not yet have Docker support on them ([INFRA-1400](https://issues.jenkins-ci.org/browse/INFRA-1400)). I guess we could have an exotic `Jenkinsfile` like

```groovy
buildPlugin(platforms: ['docker']) // usual stuff, but only on Linux
node('windows') {
    unarchive '…' // get the *.hpi
    bat 'unzip …' // extract Win32 launcher from plugin build somewhere into target/classes/…
    bat 'mvn test'
}
```

Reply:
decided to use maven build profiles to build the binaries only on the docker slave.  seemed simpler (at least in my head)